pyccel/parser/semantic.py

```diff
@@ -253,6 +254,14 @@ def annotate(self, **settings):
 
         return ast
 
+    def _get_new_name(self, current_name):
+        while current_name in self._possible_names:
+            # Generate random name. Remove dummy
+            new_suffix = create_variable(current_name).name[5:]
```

Comment:
If all we are doing here is to create a random suffix, maybe we should create a new function just for that, which would be used here and by `create_variable`

Reply:
I agree. There are also several occasions in the code where `create_variable` is used to get a name but the Symbol that it returns is rejected. I think we should create a module to handle all naming problems as discussed in issue #227 . This would be the obvious place to put this function. I thought it was better to do a pull request just for name problems which would clear up these issues. Would you rather we take care of it here?

Reply:
Solving issue #227 requires more thinking, but here we could maybe take a first step and separate the creation of a new dummy variable from the generation of a random suffix, just for clarity. @saidctb What do you think?

Reply:
Actually it's not as bad as I thought. Grep pulls up just 3 cases that need correcting

- codegen/printing/fcode.py:2338:                var_name = create_variable(r).name
- codegen/printing/fcode.py:2524:                var_name = create_variable(r).name
- parser/semantic.py:260:            new_suffix = create_variable(current_name).name[5:]